scripts/qiita_test_install

```diff
@@ -0,0 +1,841 @@
+        command = "R --version | grep 'R version' | awk '{print $3}'"
+        proc = Popen(command, shell=True, universal_newlines=True,
+                     stdout=PIPE, stderr=STDOUT)
+        stdout = proc.stdout.read()
+        version_string = stdout.strip()
+        try:
+            version = tuple(map(int, version_string.split('.')))
+            pass_test = False
+            if version[0] == minimum_version[0]:
+                if version[1] == minimum_version[1]:
+                    if version[2] >= minimum_version[2]:
+                        pass_test = True
+                elif version[1] > minimum_version[1]:
+                    pass_test = True
+            elif version[0] > minimum_version[0]:
+                pass_test = True
+        except ValueError:
+            pass_test = False
+            version_string = stdout
+        self.assertTrue(pass_test,
+                        "Unsupported R version. %s or greater is required, but running %s."
+                        % ('.'.join(map(str, minimum_version)), version_string))
+
+    def test_gdata_install(self):
+        """gdata is installed"""
+        # We currently can't programmatically find the version of gdata. An
+        # issue has been created alerting the gdata devs.
+        pass_test = True
+        try:
+            import gdata
+        except ImportError:
+            pass_test = False
+        self.assertTrue(pass_test, "gdata is not installed.")
+
+    def test_h5py(self):
+        """h5py is installed"""
+        self.assertTrue(h5py_lib_version != "Not installed.",
+                        "h5py is not installed. You should install this for "
+                        "improved performance with large BIOM files or if "
+                        "working with BIOM format version 2.x files. For "
+                        "more information, see http://biom-format.org.")
+        # cd-hit does not have a version print in their program
+
+
+def test_Qiita_config_variable(variable, Qiita_config, test,
+                               access_var=R_OK, fail_on_missing=False):
+    """test if a variable is set and set to a readable path."""
+
+    fp = Qiita_config[variable]
+
+    if not fp:
+        if fail_on_missing:
+            test.fail("%s not set." % variable)
+        else:
+            # non-essential file, so do not fail
+            return
+    # test if file exists
+    test.assertTrue(exists(fp), "%s set to an invalid file path: %s" %
+                    (variable, fp))
+
+    modes = {R_OK: "readable",
+             W_OK: "writable",
+             X_OK: "executable"}
+    # test if file readable
+    test.assertTrue(access(fp, access_var),
+                    "%s is not %s: %s" % (variable, modes[access_var], fp))
+
+
+def main():
+    option_parser, opts, args = parse_command_line_parameters(**script_info)
+
+    if opts.haiku:
+        print "Qiita provides insight\nmicrobial in nature\nto ecology"
```

Comment:
A haiku doesn't actually have an enforced syllabic structure, but the
abused English form does. It is 5-7-5 though for that form though
On Jan 20, 2015 9:21 PM, "Joshua Shorenstein" <notifications@github.com>
wrote:

> In scripts/qiita_test_install
> <https://github.com/biocore/qiita/pull/805#discussion-diff-23252284>:
>
> > +    test.assertTrue(exists(fp), "%s set to an invalid file path: %s" %
> > +                    (variable, fp))
> > +
> > +    modes = {R_OK: "readable",
> > +             W_OK: "writable",
> > +             X_OK: "executable"}
> > +    # test if file readable
> > +    test.assertTrue(access(fp, access_var),
> > +                    "%s is not %s: %s" % (variable, modes[access_var], fp))
> > +
> > +
> > +def main():
> > +    option_parser, opts, args = parse_command_line_parameters(**script_info)
> > +
> > +    if opts.haiku:
> > +        print "Qiita provides insight\nmicrobial in nature\nto ecology"
>
> 6-7-6, should be 5-7-5 [image: :tokyo_tower:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/805/files#r23252284>.
>